app/design-system/design-system-hedvig/src/main/kotlin/com/hedvig/android/design/system/hedvig/Tab.kt

```diff
@@ -0,0 +1,396 @@
+  )
+  val indicatorWidth: Dp by animateDpAsState(
+    targetValue = calculateIndicatorWidth(currentWidthMap, selectedTabIndex, density),
+    animationSpec = tween(easing = LinearEasing),
+  )
+  val indicatorHeight: Dp by animateDpAsState(
+    targetValue = calculateIndicatorHeight(currentHeightMap, selectedTabIndex, density),
+    animationSpec = tween(easing = LinearEasing),
+  )
+  Box(
+    modifier = modifier
+      .clip(tabSize.rowShape)
+      .background(tabStyle.colors.tabRowBackground)
+      .height(intrinsicSize = IntrinsicSize.Min)
+      .padding(tabSize.rowPadding),
+  ) {
+    TabIndicator(
+      indicatorHeight = indicatorHeight,
+      indicatorWidth = indicatorWidth,
+      indicatorOffsetX = indicatorOffsetX,
+      indicatorOffsetY = indicatorOffsetY,
+      indicatorColor = tabStyle.colors.chosenTabBackground,
+      indicatorShape = tabSize.tabShape,
+    )
+    FlowRow(
+      horizontalArrangement = Arrangement.Start,
+      verticalArrangement = Arrangement.Center,
+      maxItemsInEachRow = maxItemsInTheRow,
+      maxLines = maxLines,
+      overflow = FlowRowOverflow.Visible,
+      modifier = Modifier
+        .clip(tabSize.rowShape)
+        .fillMaxWidth(),
+    ) {
+      tabTitles.forEachIndexed { index, title ->
+        TabItem(
+          modifier = Modifier
+            .clip(tabSize.tabShape)
+            .weight(1f)
+            .onPlaced {
+              currentOffsetMap[index] = it.positionInParent()
+            },
+          onClick = {
+            onTabChosen(index)
+          },
+          text = title,
+          textStyle = tabSize.textStyle,
+          tabTextColor = tabStyle.colors.textColor,
+          contentPadding = tabSize.tabPadding,
+          onTextMeasured = {
+            currentWidthMap[index] = it
+          },
+          onHeightMeasured = {
+            currentHeightMap[index] = it
+          },
+        )
+      }
+    }
+  }
+}
+
+private fun calculateIndicatorWidth(map: SnapshotStateMap<Int, Int>, selectedTabIndex: Int, density: Density): Dp {
+  return with(density) {
+    map[selectedTabIndex]?.toDp() ?: 5.dp
+  }
+}
+
+private fun calculateIndicatorHeight(map: SnapshotStateMap<Int, Int>, selectedTabIndex: Int, density: Density): Dp {
+  return with(density) {
+    map[selectedTabIndex]?.toDp() ?: 5.dp
+  }
+}
+
+private fun calculateIndicatorOffsetX(map: SnapshotStateMap<Int, Offset>, selectedTabIndex: Int, density: Density): Dp {
+  val result = map[selectedTabIndex]?.x ?: 0f
+  return with(density) {
+    result.toDp()
+  }
+}
+
+private fun calculateIndicatorOffsetY(map: SnapshotStateMap<Int, Offset>, selectedTabIndex: Int, density: Density): Dp {
+  val result = map[selectedTabIndex]?.y ?: 0f
+  return with(density) {
+    result.toDp()
+  }
+}
+
+object TabDefaults {
+  internal val defaultSize: TabSize = Mini
+  internal val defaultStyle: TabStyle = Default
+  internal const val maxItemsInTheRow = 3
+  internal const val maxLines = 1
```

Comment:
Are these used anywhere else or can they be private?

Reply:
They could be out of the object and private in the file, then. I'll do that

Reply:
Could also be as part of the tokens too tbh. I put all of those in the tokens for the loading indicators https://github.com/HedvigInsurance/android/blob/4f8858ba42fb4f99b6cafd774d7d2e36cfb232d7/app/design-system/design-system-hedvig/src/main/kotlin/com/hedvig/android/design/system/hedvig/tokens/ProgressIndicatorTokens.kt#L8-L60 and it felt just fine tbh. A central spot to be able to adjust the component if need be.